### ISSUE TITLE
GCRS to Al El

### ISSUE THREAD

**Issue by AshHarvey** [astropy/astropy#10407]
### Description

I was writing some test cases for my ERFA based conversions and found a curious case when compared against AstroPy. I am using AstroPy's own ERFA codes for a lot of the transformation work. I am trying to take an Observer at a Lat, Lon, and Height to find the Az El Dist (no atmospheric effects) to an object in GCRS Cartesian coordinates. Each sub-function checks out with only very slight errors (single digit meters, 10^-16 radians), but the end to end shows a substantial (100ish Arc Seconds) error. I put test cased illustrating all of this in the test file at this [repo](https://github.com/AshHarvey/astro-trans).

I suspect the error is on my side, but tests 1-5 seems to indicate all my sub functions agree. I must be missing something in the end to end case... I am not an astrophysics guy, so I will defer to those more experienced here.

If someone has another way to check ITRS / ITRS to Az El Dist, I would be interested in cross comparing.

May relate to:
Feature Request: https://github.com/astropy/astropy/issues/10239
Issue: https://github.com/astropy/astropy/issues/4269

### Expected behavior

I would expect arc second to sub-arc second agreement.

### Actual behavior

747 mean arc second az error across 8 test points, 347 mean arc second error in el, and distance errors nearing 1 meter or less

### Steps to Reproduce

1. Clone GitHub [repo](https://github.com/AshHarvey/astro-trans)
2. Run test.py with Numpy, SciPy, AstroPy, PoliAstro, datetime, numba, and pandas installed (conda versions)

```python
Running Test Cases...
Test 1a: GCRS to ITRS (a) error in meters:  1.030988915800938
Test 1b: GCRS to ITRS (b) error in meters:  0.01145123417675297
Test 2a: ITRS to LLA error in rads,rads,meters:  [-2.22044605e-16  0.00000000e+00  0.00000000e+00]
Test 2b: ITRS to LLA (python) error in rads,rads,meters:  [-2.22044605e-16  0.00000000e+00  0.00000000e+00]
Test 3: ITRS-ITRS to Az, El, Srange error in rads,rads,meters:  [-2.22044605e-16  0.00000000e+00  0.00000000e+00]
Test 4a: Time to generate cel2ter transformation matrix with gcrs2irts_matrix_a for every 30 seconds for an entire day:  0.6632740497589111  seconds
Test 4b: Time to generate cel2ter transformation matrix with gcrs2irts_matrix_b for every 30 seconds for an entire day:  1.444657564163208  seconds
Test 5a: Cel2Ter06aXY vs Cel2Ter94, magnitude of error:  1.3718761050756562e-35
Test 5b: Cel2Ter06aXY vs Cel2Ter00aCIO, magnitude of error:  2.6046921608080613e-40
Test 5c: Cel2Ter06aXY vs Cel2Ter00aEB, magnitude of error:  4.750728822192847e-41
Test 5d: Cel2Ter06aXY vs Cel2Ter06aCA, magnitude of error:  1.6473953332649188e-39
Test 5e: Cel2Ter06aXY vs gcrs2irts_matrix, magnitude of error:  -9.44617802487642e-31
Test 5f: Cel2Ter06aXY vs utc2cel06acio, magnitude of error:  3.862790857399141e-33
Test 6a: hx error in azimuth (arc seconds) =  746.7940289878043  (mean); individual errors:  [  87.17721775 -430.92496454 2443.70246198   48.11032175 2397.34456761
 -445.32633498   44.10594255   77.66042074]
Test 6b: hx error in elevation (arc seconds) =  346.9254508681045  (mean); individual errors:  [-422.3207465   233.42024989   81.61080692  650.30147915  -82.01287173
 -233.13192588 -650.33271     422.27281687]
Test 6c: hx error in slant range (meters) =  0.3068265716483194  (mean); individual errors:  [ 0.6012234  -0.0125398   0.04346908 -0.56817969 -0.04471639  0.01091229
  0.57102116 -0.60255075]
I am unsure if these errors are mine or AstroPy's given the above matched. Each sub-function checks out against AstroPy, but the end to end case has significantly more error...
Done

Process finished with exit code 0
```

### System Details

```
Linux-5.3.0-53-generic-x86_64-with-debian-buster-sid
Python 3.7.6 | packaged by conda-forge | (default, Mar 23 2020, 23:03:20) 
[GCC 7.3.0]
Numpy 1.18.4
astropy 4.0.1.post1
Scipy 1.4.1
```

**Comment by AshHarvey** [comment]
I was able to resolve this, the error was mine. It was in the handling of lat / lon coordinates. Updated errors are:

Test 10a: hx error in azimuth (arc seconds) =  0.05567854570721198
Test 10b: hx error in elevation (arc seconds) =  0.04228777958561285
Test 10c: hx error in slant range (meters) =  -0.006513677537441254

My solution is approximately 1-2 orders of magnitude faster if anyone needs that speed up for their application.
